[Charisma Kausar] iP

ckcherry23 opened the pull request:
# Heylo

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Heylo frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/ckcherry23/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Adding priorities
- [x] User-friendly GUI 👩🏽‍💻


If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

ckcherry23 committed:
Ignore .class files

ckcherry23 committed:
Customize chatbot name

ckcherry23 committed:
Echo commands or exit program

ckcherry23 committed:
Create class to run commands

ckcherry23 committed:
Correct capitalization of bye command

ckcherry23 committed:
Ignore empty commands

ckcherry23 committed:
Add tasks to list

ckcherry23 committed:
Show list of added items

ckcherry23 committed:
Mark/Unmark tasks as done in list

ckcherry23 committed:
Remove getDescription function

ckcherry23 committed:
Add todos, events and deadlines to task list

ckcherry23 committed:
Test greeting and exit

ckcherry23 committed:
Test adding tasks and marking as done

ckcherry23 committed:
Handle missing argument exceptions

ckcherry23 committed:
Test no argument commands

ckcherry23 committed:
Convert Task array to ArrayList

ckcherry23 committed:
Delete items from task list

ckcherry23 committed:
Delete items from task list

ckcherry23 committed:
Merge branch 'master' of https://github.com/ckcherry23/ip

ckcherry23 committed:
Fix switch case formatting

ckcherry23 committed:
Close scanner on program exit

ckcherry23 committed:
Explicitly import scanner

ckcherry23 committed:
Create Tasks folder and refactor

ckcherry23 committed:
Rename package to follow java conventions

ckcherry23 committed:
Fix space formatting

ckcherry23 committed:
Create and open file for reading task list

ckcherry23 committed:
Handle case of missing data directory

ckcherry23 committed:
Ignore data directory

ckcherry23 committed:
Save tasks on adding to list

ckcherry23 committed:
Save all tasks before program exit

ckcherry23 committed:
Read saved tasks from file

ckcherry23 committed:
Check validity of dates

ckcherry23 committed:
Merge branch 'branch-Level-7'

ckcherry23 committed:
Merge branch 'branch-Level-8'

ckcherry23 committed:
Format dates using util package

ckcherry23 committed:
Handle IndexOutOfBoundsException for commands

ckcherry23 committed:
Create heylo package

ckcherry23 committed:
Change folders in text-ui-test

ckcherry23 committed:
Test toString method of Todo

ckcherry23 committed:
Test toString method of Event

ckcherry23 committed:
Test toString method of Deadline

ckcherry23 committed:
Create JAR file for App

ckcherry23 committed:
Add javadocs for Main class

ckcherry23 committed:
Add javadocs for Command class

ckcherry23 committed:
Add javadocs for FileReader class

ckcherry23 committed:
Add javadocs for FileWriter class

ckcherry23 committed:
Add javadocs for Deadline class

ckcherry23 committed:
Add javadocs for Event class

ckcherry23 committed:
Add javadocs for Task class

ckcherry23 committed:
Add javadocs for Todo class

ckcherry23 committed:
Add javadocs for DateFormatter class

ckcherry23 committed:
Format code to follow coding standard

ckcherry23 committed:
Find substrings in task list

ckcherry23 committed:
Merge branch 'branch-A-JavaDoc'

ckcherry23 committed:
Merge branch 'branch-A-CodingStandard'

ckcherry23 committed:
Merge branch 'branch-Level-9'

ckcherry23 committed:
Fix new line formatting

myc37 commented on src/main/java/heylo/Main.java:
```diff
@@ -0,0 +1,45 @@
+package heylo;
+
+import heylo.commands.Command;
```
Suggestion: The Checkstyle plugin for Intellij orders standard java package imports before other imports. It may be useful to follow this standard so that your import ordering can be standardized with your teammates for tp 😄

myc37 commented on src/main/java/heylo/commands/Command.java:
```diff
@@ -0,0 +1,107 @@
+            return;
+        }
+
+        switch (command) {
```
Good job following the correct switch-case styling!

myc37 commented on src/main/java/heylo/fileHandling/FilesReader.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * Handles reading from files.
+ */
+public class FilesReader {
```
Suggestion: FileReader might be a better name than FilesReader since this class only reads one file at a given time. Similar comment for FileWriter as well.

myc37 commented on src/main/java/heylo/tasks/Deadline.java:
```diff
@@ -0,0 +1,38 @@
+     * Create a task with a deadline.
+     *
+     * @param description String description of the task.
+     * @param deadline    String date of the deadline.
```
There's some unnecessary spacing between "deadline" and "String date of the deadline." Same comment for Event constructor.

myc37 commented on src/main/java/heylo/commands/Command.java:
```diff
@@ -0,0 +1,107 @@
+    /**
+     * Parses the command input by the user.
+     *
+     * @param input String input in the command line
```
Don't forget your full stops! 🤣

myc37 commented on src/main/java/heylo/tasks/Task.java:
```diff
@@ -0,0 +1,224 @@
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                FilesWriter.writeToFile("E : " + (event.isDone ? "1 : " : "0 : ") + event.description + " : " + event.duration + "\n");
+            } else if (task instanceof Deadline) {
```
Line 135 needs to be wrapped!

myc37 reviewed (commented):
Looks good!  Just a few suggestions to consider and nitpicks to clean up 😄

Vishwanth2210 commented on src/main/java/heylo/commands/Command.java:
```diff
@@ -0,0 +1,107 @@
+package heylo.commands;
+
+import heylo.tasks.Deadline;
+import heylo.tasks.Event;
+import heylo.tasks.Task;
+import heylo.tasks.Todo;
+
+import java.util.Scanner;
```
Nice formatting of the import statements.

Vishwanth2210 replied:
Nice formatting of the import statements.

Vishwanth2210 commented on src/main/java/heylo/commands/Command.java:
```diff
@@ -0,0 +1,107 @@
+        String[] inputArgs = input.trim().split(" ", 2);
+        String[] extraArgs = input.trim().split("/", 2);
+        this.command = inputArgs[0];
+        if (inputArgs.length > 1) {
+            this.argument = inputArgs[1];
+            if (extraArgs.length > 1 && extraArgs[0].length() > 0) {
+                this.argument = inputArgs[1].substring(0, inputArgs[1].indexOf("/"));
+                if (extraArgs[1].trim().split(" ", 2).length > 1) {
+                    this.extraInfo = extraArgs[1].trim().split(" ", 2)[1];
+                }
+            }
+        }
+    }
```
It would be better if you can avoid deep nesting.

Vishwanth2210 commented on src/main/java/heylo/tasks/Task.java:
```diff
@@ -0,0 +1,224 @@
+     * @param task Task added by user.
+     */
+    public static void addToList(Task task) {
+        if (task.description == null) {
+            System.out.println(" Please enter the description as well!");
+            System.out.println(" Command format: task-type task-description /by task-duration OR /at task-deadline");
+            return;
+        }
+
+        if (task instanceof Event) {
+            Event event = (Event) task;
+            if (event.duration == null) {
+                System.out.println(" Please enter the duration as well!");
+                System.out.println(" Command format: task-type task-description /by YYYY-MM-DD");
+                return;
+            }
+        }
+
+        if (task instanceof Deadline) {
+            Deadline taskWithDeadline = (Deadline) task;
+            if (taskWithDeadline.deadline == null) {
+                System.out.println(" Please enter the deadline as well!");
+                System.out.println(" Command format: task-type task-description /at YYYY-MM-DD");
+                return;
+            }
+        }
```
Its nice that you made the happy path prominent.

Vishwanth2210 replied:
Its nice that you made the happy path prominent.

Vishwanth2210 commented on src/main/java/heylo/tasks/Task.java:
```diff
@@ -0,0 +1,224 @@
+                FilesWriter.writeToFile("T : " + (todo.isDone ? "1 : " : "0 : ") + todo.description + "\n");
+            } else if (task instanceof Event) {
+                Event event = (Event) task;
+                FilesWriter.writeToFile("E : " + (event.isDone ? "1 : " : "0 : ") + event.description + " : " + event.duration + "\n");
```
Perhaps a line break here? 

Additionally, it would be better if you can avoid complicated expressions.

Vishwanth2210 replied:
Perhaps a line break here? 

Additionally, it would be better if you can avoid complicated expressions.

Vishwanth2210 reviewed (commented):
Good job following the coding standards. Just a few nits here and there that need to be fixed, such as line breaks and etc.

Vishwanth2210 commented on src/main/java/heylo/commands/Command.java:
```diff
@@ -0,0 +1,107 @@
+        String[] inputArgs = input.trim().split(" ", 2);
+        String[] extraArgs = input.trim().split("/", 2);
+        this.command = inputArgs[0];
+        if (inputArgs.length > 1) {
+            this.argument = inputArgs[1];
+            if (extraArgs.length > 1 && extraArgs[0].length() > 0) {
+                this.argument = inputArgs[1].substring(0, inputArgs[1].indexOf("/"));
+                if (extraArgs[1].trim().split(" ", 2).length > 1) {
+                    this.extraInfo = extraArgs[1].trim().split(" ", 2)[1];
+                }
+            }
+        }
```
It would be better if you can avoid deep nesting.

Vishwanth2210 commented on src/main/java/heylo/tasks/Task.java:
```diff
@@ -0,0 +1,224 @@
+    }
+
+    /**
+     * Prints the list of tasks in the task list.
```
Perhaps you can include the parameter subString description here?

Vishwanth2210 reviewed (commented):
Good job following the coding standards. Just need to fix a few nits such as line breaks and comments.

ckcherry23 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

ckcherry23 committed:
Merge branch 'branch-A-Gradle'

ckcherry23 committed:
Fix issues from checkstyleMain

ckcherry23 committed:
Fix issues from checkstyleTest

ckcherry23 committed:
Rename class name in build.gradle

ckcherry23 committed:
Follow SLAP in Main class

ckcherry23 committed:
Simplify toString method of tasks

ckcherry23 committed:
Simplify strings printed in Task

ckcherry23 committed:
Remove deep nesting

ckcherry23 committed:
Merge pull request #1 from ckcherry23/branch-A-CodeQuality

Follow code quality standards

ckcherry23 committed:
Fix indexOutOfBounds bug due to regressions

While improving code quality, some boolean statements were extracted
out.

This led to the length of a field being checked before checking for
its existence.

Defining the length checking boolean field in the smallest possible
scope removes this bug.

Let's move the boolean field to the inner if statement to fix the bug.

This will help us maintain bug-free code.

ckcherry23 committed:
Create FXML files from JavaFX tut part 4

ckcherry23 committed:
Update build.gradle with javafx dependencies

ckcherry23 committed:
Add images for chatbot

ckcherry23 committed:
Create GUI for the Heylo chatbot

The current implementation uses a CLI-based chatbot program
without any user interface.

Let's modify the application to include a graphical user interface
(GUI) that the user can interact with.

This will make the Heylo chatbot more user-friendly.

ckcherry23 committed:
Merge branch 'branch-Level-10'

ckcherry23 committed:
Fix import ordering

ckcherry23 committed:
Add assertions in Command class

ckcherry23 committed:
Add assertions in Task class

ckcherry23 committed:
Merge pull request #3 from ckcherry23/branch-A-Assertions

Add assertions to code

ckcherry23 committed:
Add priority level to tasks in list

Let's allow users to add priority levels among:
* NONE
* LOW
* MEDIUM
* HIGH
to their tasks in the list.

This makes the chatbot more user-friendly.

ckcherry23 committed:
Merge pull request #4 from ckcherry23/branch-C-Priority

Add priority level to tasks in list

ckcherry23 committed:
Add Ui.png

ckcherry23 committed:
Set theme jekyll-theme-tactile

ckcherry23 committed:
Remove stylized logo comment

ckcherry23 committed:
Merge branch 'master' of https://github.com/ckcherry23/ip

ckcherry23 committed:
Add user guide

ckcherry23 committed:
Set theme jekyll-theme-slate

ckcherry23 committed:
Add divider after FAQ in user guide

ckcherry23 committed:
Merge branch 'master' of https://github.com/ckcherry23/ip

ckcherry23 committed:
Update README.md

ckcherry23 committed:
Test run command function

ckcherry23 committed:
Merge branch 'master' of https://github.com/ckcherry23/ip

ckcherry23 committed:
Update user guide

ckcherry23 committed:
Remove FAQ

ckcherry23 committed:
Add table of contents to user guide

ckcherry23 committed:
User Guide: Fix toc links

ckcherry23 committed:
User Guide: Add examples for command usage

ckcherry23 committed:
User Guide: Add examples for command usage

ckcherry23 committed:
Change main class name in build.gradle

ckcherry23 committed:
Merge branch 'master' of https://github.com/ckcherry23/ip
